tests/grid/grid_generator_marching_cube_algorithm_01.cc

```diff
@@ -0,0 +1,210 @@
+  value(const Point<dim> &p, const unsigned int component) const
+  {
+    const double v = p.distance(Point<dim>()) - 0.5;
+
+    if (v > +0.1)
+      return +0.1;
+
+    if (v < -0.1)
+      return -0.1;
+
+    return v;
+  }
+};
+
+
+
+template <typename MeshType>
+MPI_Comm
+get_mpi_comm(const MeshType &mesh)
+{
+  const auto *tria_parallel = dynamic_cast<
+    const parallel::TriangulationBase<MeshType::dimension,
+                                      MeshType::space_dimension> *>(
+    &(mesh.get_triangulation()));
+
+  return tria_parallel != nullptr ? tria_parallel->get_communicator() :
+                                    MPI_COMM_SELF;
+}
+
+
+
+template <int dim, int spacedim>
+std::shared_ptr<const Utilities::MPI::Partitioner>
+create_partitioner(const DoFHandler<dim, spacedim> &dof_handler)
+{
+  IndexSet locally_relevant_dofs;
+
+  DoFTools::extract_locally_relevant_dofs(dof_handler, locally_relevant_dofs);
+
+  return std::make_shared<const Utilities::MPI::Partitioner>(
+    dof_handler.locally_owned_dofs(),
+    locally_relevant_dofs,
+    get_mpi_comm(dof_handler));
+}
+
+
+
+template <int dim>
+void
+test(const unsigned int n_subdivisions, const double iso_level)
+{
+  const unsigned int spacedim = dim + 1;
+
+  const unsigned int fe_degree      = 3;
+  const unsigned int mapping_degree = fe_degree;
+  const unsigned int n_refinements  = 5;
+
+
+  const unsigned int background_fe_degree = 3;
+
+  parallel::shared::Triangulation<spacedim> background_tria(
+    MPI_COMM_WORLD, Triangulation<spacedim>::none, true);
+
+  if (spacedim == 2)
+    GridGenerator::subdivided_hyper_rectangle(background_tria,
+                                              {20, 20},
+                                              {-1.0, -1.0},
+                                              {+1.0, +1.0});
+  else
+    GridGenerator::subdivided_hyper_rectangle(background_tria,
+                                              {20, 20, 20},
+                                              {-1.0, -1.0, -1.0},
+                                              {+1.0, +1.0, +1.0});
+
+  FE_Q<spacedim>       background_fe(background_fe_degree);
+  DoFHandler<spacedim> background_dof_handler(background_tria);
+  background_dof_handler.distribute_dofs(background_fe);
+
+  MappingQ1<spacedim> background_mapping;
+
+  VectorType ls_vector(create_partitioner(background_dof_handler));
+
+  VectorTools::interpolate(background_mapping,
+                           background_dof_handler,
+                           LSFunction<spacedim>(),
+                           ls_vector);
+
+  ls_vector.update_ghost_values();
+
+  parallel::shared::Triangulation<dim, spacedim> tria(
+    MPI_COMM_WORLD, Triangulation<dim, spacedim>::none, true);
+
+  GridGenerator::create_triangulation_with_marching_cube_algorithm(
+    background_mapping,
+    background_dof_handler,
+    ls_vector,
+    iso_level,
+    n_subdivisions,
+    tria);
+
+  // write computed vectors to Paraview
+  if (true)
```

Comment:
Can you give these `true/false` a name. This is very confusing to read.

Reply:
OK